### ISSUE TITLE
Install Plugins

### ISSUE THREAD

**Issue by blaine07** [ronivay/XenOrchestraInstallerUpdater#37]
I am a idiot; this is my own doing...I installed Xen using this excellent script. Didn't realize in the config that plugins needed to be toggled on...I have since toggled them on. Using script to update Xen doesn't seem to be installing the plugins. Is their a way I can backup, re-install and pull config back over to end up with plugins? Inside Xen under Settings>Config I see a config export; would that backup all the "Remotes" and etc fully so that I could use this script to reinstall, get plugins and import config back in? Can you elaborate on exactlly what config export in Xen does back up?

Sorry, wasn't sure how else to reach out, SORRY, I know this isn't exactly the right spot but... :-(

TLDR; didn't install with plugins, can't figure out easiest way to get plugins and not lose Xen setup in process.

Thanks, thanks so much!

**Comment by ronivay** [comment]
Hello,

How much was there time between the initial installation and the one where you tried to update with plugins? Installation doesn't continue if the installed version and currently available sources version is the same, this should be said clearly in the output though. If the base installation updates and plugins are enabled in configuration, then running the update should be enough.

If installation didn't stop because of installed and newest versions are the same, i'd love to see some output of the installation if the plugin installation failed for some reason.

Config export should export any custom configuration you've made to xen orchestra, so basically everything you need. 

Shutting down xo-server, renaming installation directory (the on you have in config file) and running install again should do the trick. Configuration is also in /var/lib/redis and /var/lib/xo-server which you can backup just in case, but don't rename. This way you most likely don't have to do any config import at all.

**Comment by blaine07** [comment]
Morning!

Yeah, I’ll wait a few more days; it’s just been a couple. If I am looking at Xen Orchestras GitHub I’m correct place a update hasn’t been made since I toggled the plugin flag to all. I’ll try to run a update in a few days/ when I see changes to source have been made. :-)

I didn’t see any errors at all; I will try to see if anything is in Xo-log when I do run update again.

So, the config export really is the key to all this? As in if I Bork Xen Orchestra it will always get me to point of last update? Your install script, too, if I understand correct makes a update of comparable nature every time it runs, too?

So, just for clarification...If in Xen Orchestra page I export config, then rename installation directory as you describe in last folder, get back to GUI and import my config back it would be 100% just how I’d left “old install”? Or I can just rename install directory defined, that’s found inside config file, re-run brand new installation it should automatically pull that setup information back over as its kept in other directory anyways? In a few days if I don’t get it to update and add plugins because sources are updated I may pursue this avenue. :-)

Thank you SO much for the help and clarification. Quite new to all this and I really appreciate your patience.

**Comment by ronivay** [comment]
> Morning!
> 
> Yeah, I’ll wait a few more days; it’s just been a couple. If I am looking at Xen Orchestras GitHub I’m correct place a update hasn’t been made since I toggled the plugin flag to all. I’ll try to run a update in a few days/ when I see changes to source have been made. :-)
>
> I didn’t see any errors at all; I will try to see if anything is in Xo-log when I do run update again.

Yup that's the easiest way to go, just wait for a commit and then update will run. It doesn't produce an error, it just says that nothing to update and quits.

> So, the config export really is the key to all this? As in if I Bork Xen Orchestra it will always get me to point of last update? Your install script, too, if I understand correct makes a update of comparable nature every time it runs, too?

Config file is readable json so you can actually export it and see what it contains. It should have all changes you've made.

> So, just for clarification...If in Xen Orchestra page I export config, then rename installation directory as you describe in last folder, get back to GUI and import my config back it would be 100% just how I’d left “old install”? Or I can just rename install directory defined, that’s found inside config file, re-run brand new installation it should automatically pull that setup information back over as its kept in other directory anyways? In a few days if I don’t get it to update and add plugins because sources are updated I may pursue this avenue. :-)

All settings (users, remote, servers and so on) are stored in /var/lib/xo-server and /var/lib/redis and this script doesn't touch those. Basically you can stop xo-server service, wipe the install directory defined in this scripts configuration and then rerun install from scratch and end up to identical configuration as long as data persists in those mentioned directories. Config backup and import is for scenarios when you don't have those directories in place anymore for some reason. Not a bad idea to do the export just in case obviously.

> Thank you SO much for the help and clarification. Quite new to all this and I really appreciate your patience.

np :) I'll close this issue for new. Feel free to reopen if you have any issues after trying the update again later on.
